ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -814,6 +814,11 @@ object Trees {
     val tpe = NoType
   }
 
+  case class TopLevelModuleExportDef(fullName: String)(
+      implicit val pos: Position) extends Tree {
+    val tpe = NoType
+  }
+
```

Comment:
I actually considered that, and took the exact opposite decision precisely because I anticipate the lazy variant to disappear. If it was going to stay, I would have taken the effort to merge the IR nodes. But given that one is going to disappear, it was much easier to create a separate IR node.

I will document the differences.